### PR TITLE
fix(commands): add delay to wait for other possible events

### LINKED ISSUE #1355: How to use VSCodeNotify() instead of VSCodeNotifyRange() after recent breaking changes with 'editor.action.reindentselectedlines' command
Previously when I called VSCodeNotifyRange had several parameters it accepted. Here is an example of how I used it before from my Lua config:

```
-- change o to also use VS Code to auto indent
normalMap('o', "o<Cmd>call VSCodeNotifyRange('editor.action.reindentselectedlines', line('.'), line('.'), 1)<CR>")
```

The idea with this command is to change the behavior of `o` to both start a new line and get VSCode to indent it.

Currently if I just change it to use `VSCodeNotify()` and only pass the command, it leaves the line unindented. If I then manually run that reindentselectedlines command, it does indent the line.

I also tried using `V` to visually select the entire line before calling `VSCodeNotify('editor.action.reindentselectedlines')`, but it still doesn't work.

On the new, unindented line, I can also manually run (via `:`) `call VSCodeNotify('editor.action.reindentselectedlines')` and it will indent the line.

### PR DESCRIPTION
**Pull request by xiyaowong**
Currently not robust

Fixes #1355

### CODE REVIEW

**Comment by xiyaowong** [comment]
@theol0403 Perhaps we can simply make VSCodeNotify deferred, as it is itself an asynchronous API

**Comment by theol0403** [comment]
@xiyaowong what is the problem with this PR, is it that adding the delay is not good for blocking user code? In that case I agree we can make it async, I thought it already was.

**Comment by xiyaowong** [comment]
When run
```lua
 vim.api.nvim_feedkeys('o', 'n', false) -- redraw
 vim.fn.VSCodeNotify('editor.action.reindentselectedlines') -- command
```
The order of events is `command`, `redraw`.

When run
```lua
  vim.api.nvim_feedkeys('o', 'n', false) -- redraw
  vim.defer_fn(function () -- command
    vim.fn.VSCodeNotify('editor.action.reindentselectedlines')
  end, 0)
```
The order of events is `redraw`, `command`.

The issue is caused by the mechanism of nvim. Add delay on vscode side isn't reliable. Therefore, it's recommended to wrap the `VSCodeNotify` to a deferred function to avoid this issue in the root.

**Comment by theol0403** [comment]
I see, thanks, that sounds good.

File: src/custom_commands_manager.ts [hunk 1]
```diff
@@ -19,8 +20,10 @@ export class CustomCommandsManager implements Disposable, NeovimCommandProcessab
     public async handleVSCodeCommand(command: string, args: unknown[]): Promise<unknown> {
         const editor = window.activeTextEditor;
         if (!editor) return;
+        // Give some time to other possible events  to weak up
+        // 10ms basically has no side effects, even `cursorMove` which may be called frequently
+        await wait(10);
```
Comment (justinmk) [inline src/custom_commands_manager.ts:25]: This is fragile and not a good tradeoff. If the goal is to wait until input is "flushed" in the Nvim server, you can do that by calling a non-fast Nvim API function, e.g. `nvim_eval('')`.

**Pull request closed without merging**